tsconsensus: add unit tests for helper functions

SilentSobs opened the pull request:
Adds non-integration unit tests to address issue #18022.

The existing tests are integration tests that spin up full tsnet.Server instances and occasionally timeout in CI. These new unit tests are fast (<10ms) and test isolated functions:

- readAllMaxBytes: tests size limiting behavior
- commandClient.url: tests URL construction
- addrFromServerAddress: tests address parsing
- raftAddr: tests raft address formatting

These tests provide baseline coverage to prevent bitrot when integration tests fail to run.

Fixes #18022

franbull reviewed (changes requested):
Thanks for your contribution! There are a couple of very minor things it would be nice to change, then we'll merge this.

franbull commented:
Please squash to one commit.

SilentSobs committed:
tsconsensus: add unit tests for helper functions

Adds non-integration unit tests to address tsconsensus needing
non-integration tests. The existing tests are integration tests that
spin up full tsnet.Server instances and occasionally timeout in CI.
These new unit tests are fast and test isolated functions:

- readAllMaxBytes: tests size limiting behavior
- commandClient.url: tests URL construction
- addrFromServerAddress: tests address parsing
- raftAddr: tests raft address formatting

Fixes #18022

Signed-off-by: Ramesh Adhikari <adhikari.resume@gmail.com>

SilentSobs commented:
> Please squash to one commit.

Done please check now.

franbull commented:
CI is complaining that there is no "signed off by" line in the commit message.  See https://github.com/tailscale/tailscale/blob/3d52c3f03e96321f1937e778e99461447b9d88dd/README.md?plain=1#L71

SilentSobs commented:
> CI is complaining that there is no "signed off by" line in the commit message. See
> 
> https://github.com/tailscale/tailscale/blob/3d52c3f03e96321f1937e778e99461447b9d88dd/README.md?plain=1#L71

signed in

franbull commented:
Looks like the CI has found some things that look pretty straightforward to address (format of test names and license headers). Are you able to click through and see the failures?

SilentSobs commented:
> Looks like the CI has found some things that look pretty straightforward to address (format of test names and license headers). Are you able to click through and see the failures?

Done

franbull commented:
`go vet` is complaining still.
